src/main/java/jenkins/branch/OrganizationFolder.java

```diff
@@ -173,7 +173,8 @@ public void complete() throws IllegalStateException, InterruptedException {
                             for (MultiBranchProjectFactory factory : projectFactories) {
                                 MultiBranchProject<?, ?> project;
                                 try {
-                                    project = factory.createProject(OrganizationFolder.this, projectName, sources, Collections.<String,Object>emptyMap(), listener);
+                                    project = factory.createProject(OrganizationFolder.this, projectName, sources, getOrphanedItemStrategy(),
```

Comment:
:ant: as noted in JIRA, there is no reason to assume that a strategy which is appropriate for _repositories_ is also appropriate for _branches_. You might be forced to retain repositories you did not really want, just in order to avoid losing branches you did want.